oci/spec_opts_windows_test.go

```diff
@@ -546,3 +548,152 @@ func TestWithImageConfigArgsEscapedWindows(t *testing.T) {
 		})
 	}
 }
+
+func TestWithWindowsDevice(t *testing.T) {
+	testcases := []struct {
+		name   string
+		IDType string
+		ID     string
+
+		expectError            bool
+		expectedWindowsDevices []specs.WindowsDevice
+	}{
+		{
+			name:        "empty_IDType_and_ID",
+			IDType:      "",
+			ID:          "",
+			expectError: true,
+		},
+		{
+			name:        "empty_IDType",
+			IDType:      "",
+			ID:          "5B45201D-F2F2-4F3B-85BB-30FF1F953599",
+			expectError: true,
+		},
+		{
+			name:   "empty_ID",
+			IDType: "class",
+			ID:     "",
```

Comment:
Why is this case valid? `class/` as a path is not enough right?

Reply:
I think it's valid at this level, since I'm trying to avoid encoding semantics about individual `IDType` values here. Docker would have let `class/` through to hcsshim, so I figured I'd keep that behaviour.

Reply:
I think we are likely to get a better error message if we validate that here. But otherwise no real issue like you say at this level. The platform will just fail later